rosette/rbl/rosette/space.rbl

```diff
@@ -86,7 +86,20 @@
     ;;; TODO: We currently do not support prologue style unification (bidirectional) pattern matching
     ;;; Potential implementation: (if (type? ptrn candidate) (seq (update) ((proc [ptrn] ptrn) candidate)) (seq (update) #niv)) )
     (pure (unify? ptrn candidate) #t)
-    (pure (match? ptrn candidate) (if (type? ptrn Symbol) #t (type? ptrn candidate)))
+    (pure (match? ptrn candidate)
+      (if (type? ptrn Symbol)
+        #t
+        (if (type? ptrn candidate)
+          (cond
+            ((type? ptrn String)
+              (= ptrn candidate))
+            ((type? ptrn Tuple)
+              (if (and (null? ptrn) (null? candidate))
+                #t
+                ((proc [[hp & ptrn] [cp & candidate]] (and (= hp cp) (match? ptrn candidate))) ptrn candidate)))
```

Comment:
Tuples have to match exactly under `=`? I would expect `['a]` and `[1]` to match given the rest of the method.

Reply:
Good catch.